### PR TITLE
Add api calls to odf

### PR DESCRIPTION
**Pull request by mrahanjam**
# ODF API

## Description
This PR adds API calls logic to odf components as well as new UI components to display pending requests (see screenshot). The only remaining part of the pending requests table is the popover which I need some clarifications from the backend team. Will add that in a separate PR as this PR has gotten big enough. Next step is to finish `create request` functionality. 

Design docs can be found [here](https://docs.google.com/document/d/1JgYSNLUZArlJ84BF8-Wiampwqde12MBTLqEjDm2pEhE/edit)

## PR Type
- [ ] Bug Fix
- [X] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement

## Links
Jira: [Jira issue #](fill-in.org)

## Test Plan
At this point, it's difficult to test as there is no way to create new requests through the UI, after that functionality is added, then testing connection requests becomes feasible. 

## Screenshots
Screen Shot 2022-01-05 at 3 51 26 PM

### CODE REVIEW

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx [hunk 1]
```diff
@@ -15,12 +15,27 @@
...
 import React from 'react';
+import { IConnection } from '../types';
+
+interface IConnectionsTableProps {
+  tableData: IConnection[];
+}
+
+const ConnectionsTable = ({ tableData }: IConnectionsTableProps) => {
+  const transformedTableData = tableData.map((req) => ({
+    requestTime: '00:00:00',
```
Comment (njbriggs) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx:26]: Should this have a date component? Also, is there a timezone?

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/index.tsx [hunk 2]
```diff
@@ -42,13 +45,40 @@ const OdfTetheringConnections = (): JSX.Element => {
...
+  const [connections, setConnections] = useState([]);
+  const [pendingRequests, setPendingRequests] = useState([]);
+
+  const fetchConnectionsList = async () => {
+    try {
+      const list = await TetheringApi.getTetheringStatusForAll().toPromise();
+      const establishedConnections = [];
+      const pendingConnections = [];
+
+      list.map((conn: IConnection) =>
```
Comment (njbriggs) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/index.tsx]: Prefer `forEach` if you're not actually returning a 1:1 map (or lodash `groupBy` for this use case).

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/index.tsx [hunk 3]
```diff
@@ -42,13 +45,40 @@ const OdfTetheringConnections = (): JSX.Element => {
...
+        conn.tetheringStatus === PENDING_STATUS
+          ? pendingConnections.push(conn)
+          : establishedConnections.push(conn)
+      );
+
+      setConnections(establishedConnections);
+      setPendingRequests(pendingConnections);
+    } catch (e) {
```
Comment (njbriggs) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/index.tsx:65]: Add a `TODO` for error messaging here?

File: app/cdap/services/helpers.js [hunk 4]
```diff
@@ -737,6 +737,10 @@ function santizeStringForHTMLID(str) {
   return str.replace(/[ \/]/g, '-');
 }
 
+function addPercentSign(str) {
```
Comment (njbriggs) [inline app/cdap/services/helpers.js]: Name nit: `formatAsPercentage`? Also this could belong in `DataFormatter`.

File: app/cdap/api/tethering.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
```
Comment (antony-britto) [inline app/cdap/api/tethering.ts]: 2022?

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx [hunk 6]
```diff
@@ -15,12 +15,27 @@
...
+    requestedResources: req.metadata.namespaceAllocations,
+  }));
 
-const ConnectionsTable = () => {
   return (
     <>
-      {/* TODO: Add table here*/}
-      <span> Table Goes Here </span>
+      {/* TODO: Will be completed after pending requests and new request creation*/}
+      {transformedTableData.map((row) => {
```
Comment (antony-britto) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx]: nit

(row) => <span> {row.gcloudProject} </span>

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx [hunk 7]
```diff
@@ -15,14 +15,145 @@
...
+          <GridCell>{isFirst ? region : ''}</GridCell>
+          <GridCell border={!isLast}>{namespace}</GridCell>
+          <GridCell border={!isLast}>{addPercentSign(cpuLimit)}</GridCell>
+          <GridCell border={!isLast}>{addPercentSign(memoryLimit)}</GridCell>
+        </GridRow>
+      );
+    });
+  };
+
+  return <>{renderTable(transformedTableData)}</>;
```
Comment (antony-britto) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx]: Do we need the fragment wrapping? <></>

File: app/cdap/api/tethering.ts (hunk 5 above)
Comment (mrahanjam) [reply app/cdap/api/tethering.ts]: updated them, good catch!

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx (hunk 1 above)
Comment (mrahanjam) [reply app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/Connections/ConnectionsTable.tsx:26]: I added a TODO here in the second commit that this needs to be properly updated once the backend server captures request time (see [this](https://cdap.atlassian.net/browse/CDAP-18717) JIRA ticket). This is more of a placeholder for now. Good point on the timezone. Need to clarify that.

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx (hunk 7 above)
Comment (mrahanjam) [reply app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx]: TIL!

File: app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx [hunk 8]
```diff
@@ -15,14 +15,147 @@
...
+  {
+    property: 'memory',
+    label: T.translate(`${PREFIX}.ColumnHeaders.memory`),
+  },
+  {
+    label: '',
+  },
+];
+
+const GridContainer = styled.div`
```
Comment (njbriggs) [inline app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx:61]: Not in this PR, but we should start using a reusable table/entity listing component. We do have https://github.com/cdapio/cdap-ui/blob/develop/app/cdap/components/shared/Table/index.tsx - did you look at that?

Comment (mrahanjam) [reply app/cdap/components/Administration/TetheringTabContent/OdfTetheringConnections/PendingRequests/PendingRequestsTable.tsx:61]: Agreed, I was just thinking that we need at least three more tables for tethering that have a lot in common. Will look into the one you linked and see how we can utilize it.

**Review by njbriggs — APPROVED** [review]
Updates to my comments look good.

**Pull request merged**